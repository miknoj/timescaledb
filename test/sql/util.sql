-- This file and its contents are licensed under the Apache License 2.0.
-- Please see the included NOTICE for copyright information and
-- LICENSE-APACHE for a copy of the license.

\set ECHO errors
\set VERBOSITY default
\c :TEST_DBNAME :ROLE_SUPERUSER

\set TMP_USER :TEST_DBNAME _wizard

DO $$
BEGIN
  ASSERT( _timescaledb_functions.get_partition_for_key(''::text) = 669664877 );
  ASSERT( _timescaledb_functions.get_partition_for_key('dev1'::text) = 1129986420 );
  ASSERT( _timescaledb_functions.get_partition_for_key('longlonglonglongpartitionkey'::text) = 1169179734);
END$$;

\pset null '[NULL]'
CREATE USER :TMP_USER;
SELECT * FROM (
    VALUES
       (_timescaledb_functions.makeaclitem(:'TMP_USER', :'TMP_USER', 'insert', false)),
       (_timescaledb_functions.makeaclitem(:'TMP_USER', :'TMP_USER', 'insert,select', false)),
       (_timescaledb_functions.makeaclitem(:'TMP_USER', :'TMP_USER', 'insert', true)),
       (_timescaledb_functions.makeaclitem(:'TMP_USER', :'TMP_USER', 'insert,select', true)),
       (_timescaledb_functions.makeaclitem(NULL, :'TMP_USER', 'insert,select', true)),
       (_timescaledb_functions.makeaclitem(:'TMP_USER', NULL, 'insert,select', true)),
       (_timescaledb_functions.makeaclitem(:'TMP_USER', :'TMP_USER', NULL, true)),
       (_timescaledb_functions.makeaclitem(:'TMP_USER', :'TMP_USER', 'insert,select', NULL)),
       (_timescaledb_functions.makeaclitem(0, :'TMP_USER', 'insert,select', true)),
       (_timescaledb_functions.makeaclitem(:'TMP_USER', 0, 'insert,select', true))
    ) AS t(item);
DROP USER :TMP_USER;
