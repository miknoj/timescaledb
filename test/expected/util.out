-- This file and its contents are licensed under the Apache License 2.0.
-- Please see the included NOTICE for copyright information and
-- LICENSE-APACHE for a copy of the license.
\set ECHO errors
                item                
------------------------------------
 db_util_wizard=a/db_util_wizard
 db_util_wizard=ar/db_util_wizard
 db_util_wizard=a*/db_util_wizard
 db_util_wizard=a*r*/db_util_wizard
 [NULL]
 [NULL]
 [NULL]
 [NULL]
 =a*r*/db_util_wizard
 db_util_wizard=a*r*/0
(10 rows)

